sdk/core/Azure.Core/src/Shared/GuidUtilities.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Azure.Core.Shared;
+
+/// <summary>
+/// The purpose of this class is to provide an allocation free way to parse and write Guids on .NET Standard 2.0, taking into account
+/// endianness.
+/// </summary>
+internal static class GuidUtilities
+{
+    private const int GuidSizeInBytes = 16;
+
+    public static Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
+    {
+        if (bytes.Length != GuidSizeInBytes)
+        {
+            throw new ArgumentException("The length of the supplied bytes must be 16.", nameof(bytes));
```

Comment:
My version had sort of a throw helper. I haven't checked but in theory it could make the code more inlineable and we might even think about adding `[MethodImpl(MethodImplOptions.AggressiveInlining)]`

Reply:
Sorry, but why would adding a method make it more inlineable?

Reply:
Ah I understand - these utility methods will be more inlineable by not throwing inline.